core/src/stored/spool.cc

```diff
@@ -492,6 +503,31 @@ static int ReadBlockFromSpoolFile(DeviceControlRecord* dcr)
   return RB_OK;
 }
 
+
+/**
+ * Despooler thread.
+ *
+ * This may or may not run the background depending on device spool
+ * configuration.
+ */
+extern "C" void* despool_thread(void* arg) {
+  struct despool_thread_args* dta = (struct despool_thread_args*)arg;
+  spool_file despool_file = dta->despool_file;
+  DeviceControlRecord* dcr = dta->dcr;
+
+  P(dcr->spool_fd_mutex[despool_file]);
+  if (!DespoolData(dcr, false, dcr->spool_fd[despool_file])) {
+    Pmsg0(000, _("Bad return from despool in WriteBlock.\n"));
+    free(arg);
+    return (void*)false;
+  }
+
+  V(dcr->spool_fd_mutex[despool_file]);
+
+  free(arg);
+  return (void*)true;
+}
```

Comment:
You can use modern C++ for this, no need for `extern "C"` and no need to pass/return a `void*`.
Also your mutex_locking can be done with `std::lock_guard` which is a lot easier to use.